add Create TSavingAccount and Update MCif

261001 opened the pull request:
![Cuplikan layar 2024-12-16 153331](https://github.com/user-attachments/assets/46b464bb-6569-4216-9cd8-01eaa86afd8a)
![Cuplikan layar 2024-12-16 201144](https://github.com/user-attachments/assets/213c916e-7217-4795-a9a9-3d998dd17daf)

Request (Put/update)
{
"customerName": "andis",
"dateOfBirth": "2007-12-14",
"address": "dago",
"phoneNumber": "09857575758787",
"email": "andis@gmail.com",
"idNumber": "7439779799777",
"idNumberType": "1"
}
Response (Put/Update)
{
    "createdBy": null,
    "createdAt": null,
    "updateBy": null,
    "updateAt": null,
    "cifId": "4028e49293c09c600193c09cdf050000",
    "customerName": "andis",
    "dateOfBirth": "2007-12-14",
    "address": "dago",
    "phoneNumber": "09857575758787",
    "email": "andis@gmail.com",
    "idNumber": "7439779799777",
    "rNumberType": {
        "typeId": "1",
        "typeCode": "0101",
        "typeName": "KTP"
    },
    "rStatus": {
        "statusId": "1",
        "statusCode": "001",
        "statusName": "Pending"
    },
    "mUserAuthorizationBy": {
        "createdBy": null,
        "createdAt": null,
        "updateBy": null,
        "updateAt": null,
        "userId": "12345",
        "isDeleted": false,
        "email": "fauziindrawan219@gmail.com",
        "userName": "fauzi",
        "password": null,
        "fullname": "fauzi ganteng",
        "token": null,
        "sessionId": null,
        "muserRole": {
            "userRoleId": "1",
            "roleCode": "1919",
            "roleName": "superAdmin",
            "deleted": false
        }
    },
    "authorization_at": null,
    "deleted": false
}

Commit:
add Create TSavingAccount
add Update MCif

xuibdg commented:
tambahkan http endpoint, dan request response nya disini

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/SimpleCoreBankingApplication.java:
```diff
@@ -2,8 +2,10 @@
+import org.springframework.context.annotation.ComponentScan;
 
 @SpringBootApplication
+@ComponentScan(basePackages = "com.b2camp.simple_core_banking")
```
Do you know the function about @ComponentScan? if you know then delete @ComponentScan

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/controller/MCifController.java:
```diff
@@ -1,18 +1,38 @@
+    @PutMapping("/{id}")
+    public ResponseEntity<?> updateCif(@PathVariable String id, @RequestBody MCif mCifEntity) {
+        MCif existingUser = mCifRepository.findById(id).orElse(null);
+        if (existingUser != null) {
```
why is there logic in the controller? please explain this is logic

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/dto/MCifResponse.java:
```diff
@@ -159,4 +160,7 @@ public String getStatusName() {
         this.statusName = statusName;
     }
+
+    public void getAuthorizationAt(Timestamp authorizationAt) {
```
why is there a param in gatter?

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/dto/TSavingAccountRequest.java:
```diff
@@ -13,19 +14,22 @@ public void setAccountNumber(String accountNumber) {
+    //getter setter untuk cifID
     public String getCifId() {
-        return cifId;
+        return cifId = cifId;
```
why cifId = cifId, are the values ​​of the two different?

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/entity/TSavingAccount.java:
```diff
@@ -163,11 +152,36 @@ public void setAuthorizationAt(LocalDateTime authorizationAt) {
         return savingId;
     }
 
     public void setSavungId(String savingId) {
```
what is savung?

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/service/MCifService.java:
```diff
@@ -1,4 +1,10 @@
+
+@Service
 public interface MCifService {
+    MCifResponse updateCif(String McifID, MCifRequest request);
```
rename param to cifId

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/service/TSavingAccountService.java:
```diff
@@ -1,4 +1,14 @@
+@Service
 public interface TSavingAccountService {
+
+    TSavingAccountResponse createe(TSavingAccountRequest tSavingAccountRequest);
```
rename method to create

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/service/impl/MCifServiceImpl.java:
```diff
@@ -1,7 +1,70 @@
+        return buildToResponseUpdate(mCif);
+    }
+
+    private MCifResponse buildToResponseUpdate(MCif mCif) {
```
rename method to buildToResponseForUpdate

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/service/impl/MCifServiceImpl.java:
```diff
@@ -1,7 +1,70 @@
+        return response;
+    }
+
+    private void buildToEntityUpdate(MCif mCif, MCifRequest request) {
```
rename method to buildToEntityForUpdate

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java:
```diff
@@ -1,6 +1,61 @@
+
+    @Transactional
+    @Override
+    public TSavingAccountResponse createe(TSavingAccountRequest tSavingAccountRequest) {
```
rename method to create

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java:
```diff
@@ -1,6 +1,61 @@
+        return buildToResponsee(tSavingAccount);
+    }
+
+    private void buildToEntityy(TSavingAccount tSavingAccount, TSavingAccountRequest request) {
```
rename to buildToEntity

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java:
```diff
@@ -1,6 +1,61 @@
+
+    }
+
+    private TSavingAccountResponse buildToResponsee(TSavingAccount tSavingAccount) {
```
rename to buildToEntity

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java:
```diff
@@ -1,6 +1,61 @@
+        tSavingAccount.setSavingId(request.getSavingId());
+        tSavingAccount.setBeginBalance(new BigDecimal("1000.00"));
+        tSavingAccount.setEndBalance(new BigDecimal("1000.00"));
+        tSavingAccount.setCurrentBalance(new BigDecimal("1000.00"));
```
BeginBalance, EndBalance and CurrentBalance, Why is the setter code not based on request?

Commit:
Already Fixing

Commit:
Merge branch 'develop' of github.com:xuibdg/simple-core-banking into CreateTSavingAccount

Commit:
Add Transactional
- isolation
- propagation
- rollback
Add Logging
- log.info
- log.error
- log.debug

Commit:
Merge remote-tracking branch 'refs/remotes/origin/develop' into CreateTSavingAccount

# Conflicts:
#	src/main/java/com/b2camp/simple_core_banking/controller/MCifController.java
#	src/main/java/com/b2camp/simple_core_banking/controller/TSavingAccountController.java
#	src/main/java/com/b2camp/simple_core_banking/dto/MCifResponse.java
#	src/main/java/com/b2camp/simple_core_banking/dto/TSavingAccountResponse.java
#	src/main/java/com/b2camp/simple_core_banking/entity/MCif.java
#	src/main/java/com/b2camp/simple_core_banking/entity/TSavingAccount.java
#	src/main/java/com/b2camp/simple_core_banking/repository/TSavingAccountRepository.java
#	src/main/java/com/b2camp/simple_core_banking/service/MCifService.java
#	src/main/java/com/b2camp/simple_core_banking/service/TSavingAccountService.java
#	src/main/java/com/b2camp/simple_core_banking/service/impl/MCifServiceImpl.java
#	src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java

Commit:
Merge remote-tracking branch 'origin/develop' into CreateTSavingAccount

Commit:
add Fixing Transactional

Commit:
add Fixing Command createdAt

Commit:
Merge remote-tracking branch 'origin/develop' into CreateTSavingAccount
